Source/AtomicJS/Javascript/JSScene.cpp

```diff
@@ -24,6 +24,27 @@ static int Node_CreateJSComponent(duk_context* ctx)
     return 1;
 }
 
+static int Node_GetJSComponent(duk_context* ctx)
+{
+	StringHash classNameHash = duk_to_string(ctx, 0);
+	duk_push_this(ctx);
+	Node* node = js_to_class_instance<Node>(ctx, -1, 0);
+
+	PODVector<JSComponent*> jsComponents;
+	node->GetComponents<JSComponent>(jsComponents);
+
+	for (unsigned i = 0; i < jsComponents.Size(); i++)
+	{
+		if (StringHash(jsComponents[i]->GetClassName()) == classNameHash)
+		{
+			js_push_class_object_instance(ctx, jsComponents[i]);
+			break;
+		}
+	}
+
+	return 1;
+}
+
```

Comment:
It would seem this should instead be a general GetComponents which takes a typehash, instead of specifically comparing the JSComponent typehash in  the function.  This would also work for JSComponents.

I filed an issue:
https://github.com/AtomicGameEngine/AtomicGameEngine/issues/110